ai/movie-recommendation/src/main/java/com/hazelcast/samples/ai/movies/TextSimilaritySearchImap.java

```diff
@@ -0,0 +1,279 @@
+ * limitations under the License.
+ */
+package com.hazelcast.samples.ai.movies;
+
+import com.hazelcast.client.HazelcastClient;
+import com.hazelcast.config.Config;
+import com.hazelcast.config.InMemoryFormat;
+import com.hazelcast.config.MapConfig;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.internal.util.Timer;
+import com.hazelcast.map.IMap;
+import com.hazelcast.query.Predicate;
+import com.hazelcast.query.Predicates;
+import dev.langchain4j.model.embedding.onnx.allminilml6v2.AllMiniLmL6V2EmbeddingModel;
+import dev.langchain4j.model.embedding.EmbeddingModel;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.regex.Pattern;
+
+/**
+ * This demo generates embeddings of movie plot summaries from the
+ * <a href="https://www.cs.cmu.edu/~ark/personas/">"CMU Movies Summary Corpus"</a>. The embeddings
+ * are stored in an IMap, along with movie metadata (title, release date).
+ * We then perform similarity search on user's input strings from console and list top 10 movies
+ * with most similar plot summaries.
+ *
+ * <p>Dependencies:</p>
+ * <ul>
+ *     <li>langchain4j for the <a href="https://huggingface.co/sentence-transformers/all-MiniLM-L6-v2">all-miniLM-L6-v2</a>
+ *     model that transforms text to 384 dimensional dense vector</li>
+ * </ul>
+ */
+public class TextSimilaritySearchImap {
+
+    static Map<String, String> movieIdToPlotSummary = new ConcurrentHashMap<>();
+    static Map<String, MovieMetadata> movieIdToMeta = new ConcurrentHashMap<>();
+
+    public static void main(String[] args) throws Exception {
+        // instantiate embedding model
+        EmbeddingModel embeddingModel = new AllMiniLmL6V2EmbeddingModel();
+
+        // start hazelcast
+        IMap<String, MovieMetadata> movies = startHzAndGetMovieImap();
+        // uncomment following line if you have a running cluster and want to execute this demo with a Hazelcast client
+//         IMap<String, MovieMetadata> movies = startHzClientAndGetMovieImap();
+        // index not necessary for a small IMap
+//        movies.addIndex(IndexType.SORTED, "releaseDate");
+
+        // ingest
+        loadSummaries(movies);
+        generateEmbeddings(movies, embeddingModel);
+
+        // read user input from console and search matches
+        try (BufferedReader consoleReader = new BufferedReader(new InputStreamReader(System.in))) {
+            while (true) {
+                // read user input
+                System.out.println("Enter release date pattern followed by search string:");
+                String line = consoleReader.readLine();
+                var parts = line.split(" ", 2);
+                if (parts.length != 2) {
```

Comment:
user input can have strings, right?
```suggestion
                if (parts.length < 2) {
```

and then take parts[1...n] as userInput, WDYT?

Reply:
the search string (`userInput`) may contain spaces